File: Frontend_QLTE/lib/models/chat_message.dart
class ChatMessage {
  final int id;
  final int adoptionRequestId;
  final String senderId;
  final String receiverId;
  final String noiDung;
  final DateTime thoiGianGui;
  final bool daDoc;

  ChatMessage({
    required this.id,
    required this.adoptionRequestId,
    required this.senderId,
    required this.receiverId,
    required this.noiDung,
    required this.thoiGianGui,
    required this.daDoc,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json["id"] ?? 0,
      adoptionRequestId: json["adoptionRequestId"] ?? 0,
      senderId: json["senderId"] ?? "",
      receiverId: json["receiverId"] ?? "",
      noiDung: json["noiDung"] ?? "",
      thoiGianGui: DateTime.tryParse(json["thoiGianGui"] ?? "") ?? DateTime.now(),
      daDoc: json["daDoc"] ?? false,
    );
  }
}

File: Backend_QLTE/Backend_QLTE/ChildService/Domain/Entities/ChatMessage.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Backend_QLTE.ChildService.Domain.Entities
{
    public class ChatMessage
    {
        [Key]
        public Guid MessageId { get; set; }

        [Required]
        public string SenderId { get; set; } = string.Empty;

        [Required]
        public string ReceiverId { get; set; } = string.Empty;

        [Required]
        public string NoiDung { get; set; } = string.Empty;

        public string? LoaiTinNhan { get; set; } // text, image, voice,...

        public DateTime ThoiGian { get; set; } = DateTime.Now;
        public bool DaDoc { get; set; } = false;
    }

}
